pkg/processor/runtime/rpc/connection/socketallocator.go

```diff
@@ -51,26 +50,37 @@ func (sa *SocketAllocator) Prepare() error {
 	if err := sa.prepareControlMessageSocket(); err != nil {
 		return errors.Wrap(err, "Failed to prepare control message socket")
 	}
+	eventSockets := make([]*EventSocket, 0)
 	for i := 0; i < sa.MinConnectionsNum; i++ {
 		eventConnection, err := sa.createSocketConnection()
 		if err != nil {
 			return errors.Wrap(err, "Failed to create event socket connection")
 		}
-		sa.eventSockets = append(sa.eventSockets,
+		eventSockets = append(eventSockets,
 			NewEventSocket(sa.Logger, eventConnection, sa))
 	}
+	// set objects in allocator
+	eventProcessors := make([]eventprocessor.EventProcessor, len(eventSockets))
+	for i, eventConnection := range eventSockets {
+		eventProcessors[i] = eventConnection
+	}
+	if err := sa.allocator.SetObjects(eventProcessors); err != nil {
+		return errors.Wrap(err, "Failed to set objects in allocator")
+	}
 	return nil
 }
 
 func (sa *SocketAllocator) Start() error {
-	if err := sa.startSockets(); err != nil {
+	eventSockets := sa.allocator.GetObjects()
+	if err := sa.startSockets(eventSockets); err != nil {
 		return errors.Wrap(err, "Failed to start socket allocator")
 	}
 
 	// wait for start if required to
 	if sa.Configuration.WaitForStart {
 		sa.Logger.Debug("Waiting for start")
-		for _, socket := range sa.eventSockets {
+		eventSockets := eventSockets
+		for _, socket := range eventSockets {
```

Comment:
why?
you don't to copy it if you're not using it in a goroutine

Reply:
Right, removing it